src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,122 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int INCREASE_RATE = 2;
+
+    private Node<K, V>[] table;
+    private int threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    private int capacity = DEFAULT_INITIAL_CAPACITY;
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> node = createNode(key, value);
+        addToTable(node, table, true);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int index = calculateIndexByHash(key == null ? 0 : key.hashCode());
+        Node<K, V> result = table[index];
+        if (result == null) {
+            return null;
+        }
+        while (result != null) {
+            if (keyEquals(result.key, key)) {
+                break;
+            }
+            result = result.next;
+        }
+        return result == null ? null : result.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K, V> createNode(K key, V value) {
+        int hashCode = (key == null) ? 0 : key.hashCode();
+        return new Node<>(hashCode, key, value, null);
+    }
+
+    private boolean keyEquals(K key1, K key2) {
+        return (key1 == key2) || (key1 != null && key1.equals(key2));
+    }
+
+    private void increaseSize(boolean increasing) {
+        if (increasing) {
+            size++;
+            if (size >= threshold) {
+                resize();
+            }
+        }
+    }
```

Comment:
boolean as method parameters is a bad practice
```suggestion
    private void increaseSize() {
            size++;
            if (size >= threshold) {
                resize();
            }
    }
```